hack/lib/util.sh

```diff
@@ -221,21 +221,6 @@ kube::util::gen-docs() {
   popd > /dev/null
 }
 
-# Puts a placeholder for every generated doc. This makes the link checker work.
```

Comment:
so this looks like the reason we've been keeping these around? wonder where/what it is

Reply:
The reason we want the link checker to continue to work is that we want a guarantee that the links to continue to work.

I'm OK if you found another way to solve the problem, but this PR needs to explain that. See also: https://en.wikipedia.org/wiki/Wikipedia:Chesterton%27s_fence

Reply:
i need to ask what is the "link checker" and what "links" are talking about?

the real docs are generated (and also HTML prettyfied) by sig-docs before a release and are published on the k8s website:
https://kubernetes.io/docs/reference/
what control does the k/k repo has over that?

i don't see the value of validating the existence of placeholders files listed in .generated_docs.
also, for the real files the list is populated based on file existence.

> Chesterton's fence is the principle that reforms should not be made until the reasoning behind the existing state of affairs is understood.

+1, let's understand all the reasons.

Reply:
bump, i need some clarifications here.
thanks.

Reply:
+1